includes/internal/emails/class-email-generator.php

```diff
@@ -69,19 +69,28 @@ public function __construct( Email_Repository $email_repository, Lesson_Progress
 	 * @internal
 	 */
 	public function init(): void {
-		$this->email_generators = [
-			Student_Starts_Course::IDENTIFIER_NAME    => new Student_Starts_Course( $this->email_repository ),
-			Student_Completes_Course::IDENTIFIER_NAME => new Student_Completes_Course( $this->email_repository ),
-			Student_Completes_Lesson::IDENTIFIER_NAME => new Student_Completes_Lesson( $this->email_repository, $this->lesson_progress_repository ),
-			Student_Submits_Quiz::IDENTIFIER_NAME     => new Student_Submits_Quiz( $this->email_repository ),
-			Course_Completed::IDENTIFIER_NAME         => new Course_Completed( $this->email_repository ),
-			New_Course_Assigned::IDENTIFIER_NAME      => new New_Course_Assigned( $this->email_repository ),
-			Quiz_Graded::IDENTIFIER_NAME              => new Quiz_Graded( $this->email_repository ),
-			Teacher_Message_Reply::IDENTIFIER_NAME    => new Teacher_Message_Reply( $this->email_repository ),
-			Student_Message_Reply::IDENTIFIER_NAME    => new Student_Message_Reply( $this->email_repository ),
-			Student_Sends_Message::IDENTIFIER_NAME    => new Student_Sends_Message( $this->email_repository ),
+		$generator_instances = [
+			new Student_Starts_Course( $this->email_repository ),
+			new Student_Completes_Course( $this->email_repository ),
+			new Student_Completes_Lesson( $this->email_repository, $this->lesson_progress_repository ),
+			new Student_Submits_Quiz( $this->email_repository ),
+			new Course_Completed( $this->email_repository ),
+			new New_Course_Assigned( $this->email_repository ),
+			new Quiz_Graded( $this->email_repository ),
+			new Teacher_Message_Reply( $this->email_repository ),
+			new Student_Message_Reply( $this->email_repository ),
+			new Student_Sends_Message( $this->email_repository ),
 		];
 
+		$this->email_generators = array_reduce(
```

Comment:
Is it creating a mapping between a key and classes?  
The refactored versions seem more complicated to read

Reply:
Thanks, I've reverted the change [here](https://github.com/Automattic/sensei/pull/6617/commits/0c02f288db784dba1d13e9037eded839633378a8)